srml/support/src/dispatch.rs

```diff
@@ -1342,6 +1416,11 @@ mod tests {
 					]),
 					documentation: DecodeDifferent::Encode(&[]),
 				},
+				FunctionMetadata {
```

Comment:
Do we need the weight in the metadata?

CC @jacogr

Reply:
no clue. I just added it to pass the test tbh and assumed it happens by default.

Reply:
If this is affecting fee calculation (down the line), it should be made available. (There is obviously a downstream effect for any metadata version bumps, which this should be if the structure changes - at least the JS and Python metadata parsers should be updated to cater for the new v5)

Reply:
@jacogr, for now, it is not affecting the fee calculation and at least in this PR I don't intend to do that but rather build the building blocks. Though, for sure, very soon the fees will be affected by the weight. Not sure what this means for js/client side.

Reply:
Ok, so then I would suggest not yet exposing the weights on the metadata. For each change on the metadata formats, we basically -

1. need to bump the metadata version
2. downstream projects need to implement new versions of the Metadata (unlike Substrate, they need to support all. This also includes stuff like Polkascan, just like the polkadotjs/api they support from v0 all the way up to current v4)

I'm not against a v5 bump now the works needs to be done at some point anyway, however if things may change again and end up with a v6, maybe not - especially not if the stuff is not used atm.

Without the metadata changes (where there go down the wire), as it stands (and yes, looked at what the code does), there is no impact as of yet. 

Excited to see how this evolves, it is a really cool approach.

Reply:
@bkchr how can one prevent this information to sneak into metadata then?